model/task_lifecycle.go

```diff
@@ -604,3 +608,56 @@ func MarkTaskDispatched(t *task.Task, hostId, distroId string) error {
 	}
 	return nil
 }
+
+// RestartFailedTasks attempts to restart failed tasks that started between 2 times
+// It returns a slice of task IDs that were successfully restarted as well as a slice
+// of task IDs that failed to restart
+func RestartFailedTasks(startTime, endTime time.Time, user string, dryRun bool) ([]string, []string, error) {
+	tasksToRestart, err := task.Find(task.ByTimeStartedAndFailed(startTime, endTime))
+	if err != nil {
+		return nil, nil, err
+	}
+	tasksRestarted := make([]string, 0)
+	var tasksErrored []string
+	if !dryRun {
+		tasksErrored = make([]string, 0)
+	}
+	for _, t := range tasksToRestart {
+		if dryRun {
+			tasksRestarted = append(tasksRestarted, t.Id)
+			continue
+		}
+		projectRef, err := FindOneProjectRef(t.Project)
+		if err != nil {
+			tasksErrored = append(tasksErrored, t.Id)
+			grip.Error(message.Fields{
+				"task":    t.Id,
+				"status":  "failed",
+				"message": fmt.Sprintf("error retrieving project ref: %v", err.Error()),
```

Comment:
I would have the error as a separate field in this message.

Reply:
done